app/logic/Mail/EWS/EWSFolder.ts

```diff
@@ -41,19 +41,20 @@ export class EWSFolder extends Folder {
     }
   }
 
-  async listMessages(): Promise<ArrayColl<EWSEMail>> {
+  async listMessages() {
     if (!this.dbID) {
       await SQLFolder.save(this);
     }
     await SQLEMail.readAll(this);
 
-    return this.updateChangedMessages();
+    this.updateChangedMessages();
```

Comment:
No, because IMAP's `listMessages` doesn't know which messages are new, instead having a separate function to list only new messages, so for consistency of the function signature I separated which functions deal with new messages and which don't.